api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java

```diff
@@ -407,24 +407,31 @@ public void sendBillingSetupEmail(DbUser dbUser, SendBillingSetupEmailRequest em
 
   @Override
   public void sendEgressRemediationEmail(
-      DbUser dbUser, EgressRemediationAction action, @Nullable String gkeServiceName)
+      DbUser dbUser,
+      EgressRemediationAction action,
+      @Nullable String gkeServiceName,
+      boolean isVwbEgress)
       throws MessagingException {
     String remediation = EGRESS_REMEDIATION_ACTION_MAP.get(action);
-    String environmentType =
-        appServiceNameToAppType(Strings.nullToEmpty(gkeServiceName))
-            .map(LeonardoAppUtils::appDisplayName)
-            .orElse("Jupyter");
-
     String givenName = Optional.ofNullable(dbUser.getGivenName()).orElse("Researcher");
 
+    String egressSource = "";
+    if (!isVwbEgress) {
+      String environmentType =
+          appServiceNameToAppType(Strings.nullToEmpty(gkeServiceName))
+              .map(LeonardoAppUtils::appDisplayName)
+              .orElse("Jupyter");
+      egressSource = String.format(EGRESS_SOURCE, environmentType);
```

Comment:
is the EGRESS_SOURCE being formatted somewhere in case it's a VWB egress?

Reply:
nope. I was thinking for now, for vwb egress, it would just be `egress from
  your workspac`without telling user app type